R/RtGam.R

```diff
@@ -83,8 +95,21 @@ RtGam <- function(cases,
       ...
     )
   )
+  diagnostics <- calculate_diagnostics(fit)
+  if (warn_for_diagnostic_failure) {
+    issue_diagnostic_warnings(diagnostics)
+  }
 
-  invisible(NULL)
+  format_for_return(
+    fit = fit,
+    df = df,
+    group = group,
+    k = k,
+    m = m,
+    backend = backend,
+    formula = formula,
+    diagnostics = diagnostics
```

Comment:
I don't feel that strongly but do you feel strongly that there is value to storing the diagnostics vs just having users call `check_diagnostics`?

Reply:
I do. I think we want the diagnostics in the returned object and `check_diagnostics()` is ancillary. I was debating whether to have `check_diagnostics()` at all.

The key point here is that I think we want to do automated checks as a first class citizen. They're a primary concern and part of any model run. Allowing the users to run them again is nice but ancillary. And if we're going to generate the diagnostics, we should at least store them.

Reply:
> Allowing the users to run them again is nice but ancillary. 

Is it nice? As in when would they do this vs look at the diagnostics in the fit output (i.e your point)

Reply:
you could have it so that this just accesses the diagnostics in the fit object for people

Reply:
https://github.com/CDCgov/cfa-gam-rt/pull/22/commits/1f9702e1f5942829f074e37939fe0256125f40b2